designdocs/provider_registration.md

```diff
@@ -0,0 +1,183 @@
+# Provider Registration
+
+## Table of Contents
+* [Context and Scope](#context-and-scope)
+* [Problem Statement](#problem-statement)
+* [Goals and Non-Goals](#goals-and-non-goals)
+* [Proposal](#proposal)
+* [Benefits and Risks](#benefits-and-risks)
+* [Alternatives and Rationale](#alternatives-and-rationale)
+* [Glossary](#glossary)
+
+## Context and Scope
+In order to properly implement MRC, we need to describe how service providers
+will participate on the network.
+
+## Problem Statement
+Service providers are an integral part of how MRC functions. However, there are
+some open questions about how they operate, questions that this document hopes
+to provide clarity on.
+
+Among those questions are:
+
+1. Who can become a provider?
+1. How does a user become a provider?
+1. What are the attributes of a provider?
+1. How will providers be identified to other network actors?
+1. How does one stop being a provider?
+1. Can provider status be revoked from a provider?
+
+## Goals and Non-Goals
+This document will discuss the ways in which service providers can register with
+and participate in the MRC network.
+
+This document will not discuss the specifics of how providers will process
+service requests from users on the network. This document will also not discuss
+specifics around creating and honoring transactions between users.
+
+## Proposal
+The basic workflow for provider registration is as follows:
+1. User stakes balance to become a provider. This is called a "Registration Fee".
+1. The "Registration Fee" is credited to the Chain Treasury (TBD).
+1. The provider's MSA id is added to storage on chain.
+
+Should a provider choose at any point to stop providing services, it can issue a
+request to remove itself from the provider registry:
+1. The provider issues an announcement that it will stop providing services.
+1. The provider's MSA id is removed from storage.
+
+
+Please note:
+* All names are placeholders and may be changed.
+* Types may change as needed during implementation phase
+* Errors in the extrinsic(s) must have different, reasonably-named error enums
+  for each type of error for ease of debugging.
+
+### Types
+* `ProviderRegistrationParams<T: Config>`, the arguments used to register an announcement.
+  * `provider_msa_id`: `MsaId`
+  * `registration_fee`: `Balance`
```

Comment:
Will this be a fixed amount? If so, we may not need to include it.

Reply:
Is this an additional fee aside from the fees calculated from weights?

Reply:
tips may,  maybe a look at crowloan pallet or something like will give us an idea of different kinds

Reply:
> Is this an additional fee aside from the fees calculated from weights?

No, this would be the actual registration fee.

Reply:
Good question. We can make it a governance managed amount. So that would remove it from the params.

Reply:
@wilwade thoughts on this language from the benefits section?

> It may be prudent, in the future, to determine whether or not
registrations are fixed or vary based on the amount of verifable information
given on a registration attempt. The latter may allow us to place more trust
into providers that do not have traditional business attributes.